packages/medusa/src/api/routes/admin/price-lists/list-price-list-products.ts

```diff
@@ -78,31 +81,79 @@ export default async (req, res) => {
 
   req.query.price_list_id = [id]
 
-  const filterableFields: FilterableProductProps = omit(req.query, [
+  const query: FilterableProductProps = omit(req.query, [
     "limit",
     "offset",
     "expand",
     "fields",
     "order",
   ])
 
-  const result = await listAndCount(
-    req.scope,
-    filterableFields,
-    {},
-    {
-      limit: validatedParams.limit ?? 50,
-      offset: validatedParams.offset ?? 0,
-      expand: validatedParams.expand,
-      fields: validatedParams.fields,
-      order: validatedParams.order,
-      allowedFields: allowedAdminProductFields,
-      defaultFields: defaultAdminProductFields as (keyof Product)[],
-      defaultRelations: defaultAdminProductRelations,
+  const limit = validatedParams.limit ?? 50
+  const offset = validatedParams.offset ?? 0
+  const expand = validatedParams.expand
+  const fields = validatedParams.fields
+  const order = validatedParams.order
+  const allowedFields = allowedAdminProductFields
+  const defaultFields = defaultAdminProductFields as (keyof Product)[]
+  const defaultRelations = defaultAdminProductRelations.filter(
+    (r) => r !== "variants.prices"
+  )
+
+  const priceListService: PriceListService =
+    req.scope.resolve("priceListService")
+
+  let includeFields: (keyof Product)[] | undefined
+  if (fields) {
+    includeFields = fields.split(",") as (keyof Product)[]
+  }
+
+  let expandFields: string[] | undefined
+  if (expand) {
+    expandFields = expand.split(",")
+  }
+
+  let orderBy: { [k: symbol]: "DESC" | "ASC" } | undefined
+  if (typeof order !== "undefined") {
+    let orderField = order
+    if (order.startsWith("-")) {
+      const [, field] = order.split("-")
+      orderField = field
+      orderBy = { [field]: "DESC" }
+    } else {
+      orderBy = { [order]: "ASC" }
+    }
+
+    if (!(allowedFields || []).includes(orderField)) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Order field must be a valid product field"
+      )
     }
+  }
+
+  const listConfig = getListConfig<Product>(
+    defaultFields ?? [],
+    defaultRelations ?? [],
+    includeFields,
+    expandFields,
+    limit,
+    offset,
+    orderBy
+  )
+
+  const [products, count] = await priceListService.listProducts(
+    id,
+    pickBy(query, (val) => typeof val !== "undefined"),
```

Comment:
suggestion(non-blocking, if-minor): extract to function

Reply:
I'll create a separate issue for this to fix it across list endpoints where we use this pattern